### DISCUSSION TITLE
Redirecting based on cookie present in next.config.js

### DISCUSSION THREAD

**Discussion by Shooshte** [vercel/next.js#37057] [Help] [answered]
Hi guys,

Our app uses a cookie names TSSO (provided by a SSO provider). If the cookie is set, this means the user is authenticated. I would like to implement a perma redirect to `/landing` if the cookie is not set inside next.config.js, but do not know how to check only if the cookie is there (as I do not know the value of the cookie set by the SSO provider).

Based on redirect docs, I got so far, but now I do not know how to turn this condition:
```
const PRIVATE_PATHS = [
  '/location',
  '/recommended_items',
  '/checkout',
  '/customer_profile/:section?',
  '/categories_with_items',
  '/categories',
  '/subcategories_with_items/:catName',
  '/subcategory_items/:catName/:subcatName',
  '/recipe/:recName',
  '/recipes',
  '/promotion/:promoName',
  '/completed-order',
];

const PRIVATE_REDIRECTS = PRIVATE_PATHS.map((path) => {
  return {
    source: path,
    has: [
      {
        type: 'cookie',
        key: 'TSSO',
      },
    ],
    destination: '/landing',
    permanent: false,
  };
});

// https://nextjs.org/docs/api-reference/next.config.js/introduction
const nextConfig = {
  async redirects() {
    return [
      {
        source: '/',
        destination: '/landing',
        permanent: true,
      },
      ...PRIVATE_REDIRECTS,
    ];
  },
  images: {
    // TODO add all image domains here
    domains: [],
  },
  reactStrictMode: true,
};

const moduleExports = nextConfig;
```

Do I need to implement this inside of my `_app.tsx` Component `getServerSideProps` and check for req cookies there or is there a way to handle it using next.config?

**Comment by icyJoseph** [comment, marked as answer]
I'd do this on `pages/_middleware.ts`, if you have one of the latest next.js versions:

```ts
import { NextResponse, type NextRequest } from "next/server";

export function middleware(req: NextRequest) {
  const { pathname } = req.nextUrl;

  if (
    pathname.startsWith("/api") || //  exclude all API routes
    pathname.startsWith("/static") || // exclude static files
    pathname.includes(".") // exclude all files in the public folder
  ) {
    return NextResponse.next();
  }

  // let this go through
  if (pathname === "/landing") return NextResponse.next();

  const ssoCookie = req.cookies["TSSO"];

  if (!ssoCookie) {
    req.nextUrl.pathname = "/landing";
    return NextResponse.redirect(req.nextUrl);
  }

  // otherwise the header is present
  return NextResponse.next();
}
```

Though, if there's a shallow link going to a page that needs the header, this won't work. Not sure if `next.config.js` works.

**Reply by Shooshte** [reply]
For my specific use case the value that I needed was the cookie: `req.cookies.TSSO`. Marking this as an answer since it enabled me to solve my problem (no shallow rendering for now).

**Reply by icyJoseph** [reply]
Ah my bad, I didn't read correctly. Will edit the answer to read the cookies. I thought you'd have a header and wrote my answer with that. Good stuff!
